src/user-lists/user-list.service.ts

```diff
@@ -309,7 +309,17 @@ export class UserListService {
     queryBuilder.orderBy("full_name", orderBy);
     queryBuilder.offset(pageOptionsDto.skip).limit(pageOptionsDto.limit);
 
-    const [itemCount, entities] = await Promise.all([queryBuilder.getCount(), queryBuilder.getMany()]);
+    const subQuery = this.userHighlightRepository.manager
+      .createQueryBuilder()
+      .from(`(${queryBuilder.getQuery()})`, "subquery_for_count")
+      .addCommonTableExpression(queryBuilder, "CTE")
+      .setParameters(queryBuilder.getParameters())
+      .select("count(full_name)");
```

Comment:
Apologies, i gave you a malformed chunk. This should work:

```suggestion
    const subQuery = this.userHighlightRepository.manager
      .createQueryBuilder()
      .addCommonTableExpression(queryBuilder, "CTE")
      .setParameters(queryBuilder.getParameters())
      .select("count(full_name)")
      .from("CTE", "CTE");
```

Reply:
The "from" should be on the `CTE` is the biggest thing thing you need with a `addCommonTableExpression`